java/core/src/main/java/com/whylogs/core/DatasetProfile.java

```diff
@@ -94,9 +99,10 @@ public void setDatasetTimestamp(Instant datasetTimestamp) {
 
   private void initializeNewColumns(Set<String> colNames) {
     for (String column : colNames) {
-      ColumnSchema columnSchema = this.schema.columns.get(column);
+      ColumnSchema columnSchema = this.schema.getColumns().get(column);
       if (columnSchema != null) {
-        this.columns.put(column, new ColumnProfile(column, columnSchema, this.schema.cache_size));
+        this.columns.put(
+            column, new ColumnProfile<>(column, columnSchema, this.schema.getCacheSize()));
```

Comment:
maybe too late to save you much here, but the cache only really makes sense in python, where there is a C++ library (whylogs-sketching) that can handle arrays of inputs more efficiently. In whylogs Java the sketching library is also Java so this looks like a premature optimization.

Reply:
Ok, given it's already here do you think we should leave it or go ahead?

Reply:
I'll take out the cache.

Reply:
Ok, well looking into this if we decide to do that then we should do that in a separate PR as it does touch quite a few things.